### ISSUE TITLE
[no-type-alias] Rule fails with "Unhandled in union types are not allowed." when encountering null

### ISSUE THREAD

**Issue by mdouglass** [typescript-eslint/typescript-eslint#2328]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-type-alias": [
      "error",
      {
        "allowAliases": "always",
        "allowCallbacks": "always",
        "allowLiterals": "in-unions-and-intersections",
        "allowMappedTypes": "always",
        "allowTupleTypes": "always",
      },
    ]
  }
}
```

```TS
export type Value = null | string | number
```

**Expected Result**
No errors

**Actual Result**
Unhandled in union types are not allowed.eslint@typescript-eslint/no-type-alias

**Versions**
| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.0` |
| `@typescript-eslint/parser`        | `3.7.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `14.6.0` |
| `npm`                              | `6.14.6` |

**Comment by bradzacher** [comment]
This is fixed by #2316. I accidentally introduced a regression in the AST.
This will be released tomorrow morning - you can try the `canary` tag on NPM in the meantime.

**Comment by mdouglass** [comment]
Awesome, thank you. I'll give it a try shortly.
